<launch>

	<!-- global arguments -->
	<!-- attacker1 joystick enable -->
	<arg name="a1_joy" default="False"/>
	<arg name="a1_joy_dev" default="/dev/input/js0"/>
	<arg name="record_bag" default="False"/>
	<arg name="topics" default="/defenders_locations /enemy_locations /takeoff /defender1/dlp_state /defender2/dlp_state /defender3/dlp_state /battle /commander /defender1/mavros/extended_state /defender2/mavros/extended_state /defender3/mavros/extended_state /attacker1/mavros/extended_state /defender1/mavros/local_position/velocity /defender2/mavros/local_position/velocity /defender3/mavros/local_position/velocity /attacker1/mavros/local_position/velocity /defender1/mavros/state /defender2/mavros/state /defender3/mavros/state /attacker1/mavros/state"/>

	<node pkg="rosbag" type="record" name="bag_recording" args="-o /home/abdkadma/ros_bags/zahana_game/simulated_game.bag $(arg topics)" required="true" if="$(arg record_bag)">
	</node>

	<!-- Launch defenders nodes-->
	<!-- defender1 -->
	<group ns="defender1">

		<arg name="id" default="0"/>
		<rosparam command="load" file="$(find dlp)/config/problem_setup.yaml" />
		<param name="myID" type="int" value="$(arg id)" />

		<node pkg="dlp" type="dlp_node" name="dlp_node_$(arg id)"  required="false" output="screen">
    </node>
		<node pkg="dlp" type="defender_node.py" name="defender_node_$(arg id)"  required="false" output="screen">
    </node>

	</group>

	<!-- launch attackers nodes -->
	<!-- attacker1 -->
	<group ns="attacker1">

		<arg name="id" default="0"/>

		<rosparam command="load" file="$(find dlp)/config/problem_setup.yaml" />

		<param name="myID" type="int" value="$(arg id)" />
		<param name="debug" value="false" />
		<param name="alpha" value="0.8" />
		<param name="beta" value="-0.2" />
		<param name="neighbor_radius" value="1" />
		<param name="Tp" value="3" />
		
		<param name="enable_joy" type="bool" value="$(arg a1_joy)" />

		<node pkg="joy" type="joy_node" name="joy_$(arg id)"  required="true" output="screen" if="$(arg a1_joy)">
			<param name="dev" type="string" value="$(arg a1_joy_dev)" />
		</node>

		<node pkg="dlp" type="attacker_node.py" name="attacker_node_$(arg id)"  required="false" output="screen">
    </node>

		<node pkg="dlp" type="dlp_node_attacking" name="dlp_node_attacking"  required="false" output="screen">
    </node>

	</group>

</launch>
